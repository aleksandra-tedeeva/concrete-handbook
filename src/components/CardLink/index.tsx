import { Info } from '@mui/icons-material';
import { Card, CardContent, Stack, Typography, styled } from '@mui/material';
import { Link } from 'react-router-dom';

export interface CardLinkProps {
  name: string;
  link: string;
}

const StyledLink = styled(Link)({
  textDecoration: 'none',
  color: 'inherit'
});

export default function CardLink({ name, link }: CardLinkProps) {
  return (
    <Card variant="outlined">
      <CardContent>
        <Stack
          spacing={2}
          alignItems="center"
          justifyContent="center"
          textAlign="center"
          sx={(theme) => ({
            textDecoration: 'none',
            color: theme.palette.secondary.main,
            height: '100px'
          })}
        >
          <StyledLink to={link}>
            <Info sx={{ width: 40, height: 40 }} />
            <Typography variant="h6">{name}</Typography>
          </StyledLink>
        </Stack>
      </CardContent>
    </Card>
  );
}
