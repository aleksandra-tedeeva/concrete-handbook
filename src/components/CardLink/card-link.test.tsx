import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import CardLink from '.';

jest.mock('react-router-dom', () => {
  const originalModule = jest.requireActual('react-router-dom');

  return {
    __esModule: true,
    ...originalModule,
    useParams: jest.fn(),
    useHistory: jest.fn()
  };
});

describe('CardLink', () => {
  it('должен правильно отображаться на странице', () => {
    const { container } = render(
      <BrowserRouter>
        <CardLink name="Тестовое Название" link="/test-link" />
      </BrowserRouter>
    );

    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('должен правильно отображать параметр `name`', () => {
    render(
      <BrowserRouter>
        <CardLink name="Тестовое Название" link="/test-link" />
      </BrowserRouter>
    );

    const name = screen.getByRole('heading');
    expect(name.textContent).toEqual('Тестовое Название');
  });

  it('должен правильно отображать параметр `link`', () => {
    render(
      <BrowserRouter>
        <CardLink name="Тестовое Название" link="/test-link" />
      </BrowserRouter>
    );

    const { href } = screen.getByRole('link') as HTMLLinkElement;
    expect(href).toEqual('http://localhost/test-link');
  });

  it('должен переходить по ссылке `link` при клике на карту', () => {
    render(
      <BrowserRouter>
        <CardLink name="Тестовое Название" link="/test-link" />
      </BrowserRouter>
    );

    const locationBeforeRouting = window.location.href;
    expect(locationBeforeRouting).toEqual('http://localhost/');

    const link = screen.getByRole('link') as HTMLLinkElement;
    fireEvent.click(link);

    const locationAfterRouting = window.location.href;
    expect(locationAfterRouting).toEqual('http://localhost/test-link');
  });
});
