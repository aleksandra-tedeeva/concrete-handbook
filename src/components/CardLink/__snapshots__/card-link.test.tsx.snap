// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardLink должен правильно отображаться на странице 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="MuiStack-root css-rehsbp-MuiStack-root"
      >
        <a
          class="css-nn640c"
          href="/test-link"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1jt09b6-MuiSvgIcon-root"
            data-testid="InfoIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
            />
          </svg>
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Тестовое Название
          </h6>
        </a>
      </div>
    </div>
  </div>
</div>
`;
